docs/general/queries.md

```diff
@@ -44,3 +44,57 @@ schema = strawberry.Schema(query=Query)
 ```
 
 So now, when requesting the name field, the `get_name` function will be called.
+
+Alternatively a field can be declared using a decorator:
+
+```python
+@strawberry.type
+class Query:
+    @strawberry.field
+    def name(self) -> str:
+        return "Strawberry"
+```
+
+## Arguments
+
+GraphQL fields can accept arguments, usually to filter out or retrieve specific
+objects:
+
+```python
+FRUITS = [
+    "Strawberry",
+    "Apple",
+    "Orange",
+]
+
+
+@strawberry.type
+class Query:
+    @strawberry.field
+    def fruit(self, startswith: str) -> str | None:
+        for fruit in FRUITS:
+            if fruit.startswith(startswith):
+                return fruit
+        return None
+```
+
+Additional metadata can be added to arguments, for example a custom name and
+description using `strawberry.argument` with
+[typing.Annotated](https://docs.python.org/3/library/typing.html#typing.Annotated):
+
+```python
+@strawberry.type
+class Query:
+    @strawberry.field
+    def fruits(
+        self,
+        is_tasty: Annotated[
+            bool | None,
+            strawberry.argument(
+                description="Filters out fruits by whenever they're tasty or not",
+                deprecation_reason="isTasty argument is deprecated, "
+                "use fruits(taste:SWEET) instead",
```

Comment:
How about 🤔
```suggestion
                deprecation_reason="isTasty argument is deprecated, use fruits(taste:SWEET) instead",
```

----
```python
@strawberry.type
class Query:
    @strawberry.field
    def fruits(
        self,
        is_tasty: Annotated[
            bool | None,
            strawberry.argument(
                description="Filters out fruits by whenever they're tasty or not",
                deprecation_reason="isTasty argument is deprecated, "
                "use fruits(taste:SWEET) instead",
            ),
        ] = None,
    ) -> list[str]: ...
```

----
```python
@strawberry.type
class Query:
    @strawberry.field
    def fruits(
        self,
        is_tasty: Annotated[
            bool | None,
            strawberry.argument(
                description="Filters out fruits by whenever they're tasty or not",
                deprecation_reason="isTasty argument is deprecated, use fruits(taste:SWEET) instead",
            ),
        ] = None,
    ) -> list[str]: ...
```

Reply:
This it a bit too long (101) characters, it's hard to read in git diffs, and black by default recommends 88-80 character limit